hsmd/hsmd.c

```diff
@@ -1525,6 +1527,107 @@ static struct io_plan *handle_sign_invoice(struct io_conn *conn,
 			 take(towire_hsm_sign_invoice_reply(NULL, &rsig)));
 }
 
+/* stolen from gossipd/routing.c to parse wireaddr in node announcement msg*/
+static struct wireaddr *read_addresses(const tal_t *ctx, const u8 *ser)
+{
+	const u8 *cursor = ser;
+	size_t len = tal_count(ser);
+	struct wireaddr *wireaddrs = tal_arr(ctx, struct wireaddr, 0);
+
+	while (cursor && len) {
+		struct wireaddr wireaddr;
+
+		/* BOLT #7:
+		 *
+		 * The receiving node:
+		 *...
+		 *   - SHOULD ignore the first `address descriptor` that does
+		 *     NOT match the types defined above.
+		 */
+		if (!fromwire_wireaddr(&cursor, &len, &wireaddr)) {
+			if (!cursor)
+				/* Parsing address failed */
+				return tal_free(wireaddrs);
+			/* Unknown type, stop there. */
+			status_trace("read_addresses: unknown address type %u",
+				     cursor[0]);
+			break;
+		}
+
+		tal_arr_expand(&wireaddrs, wireaddr);
+	}
+	return wireaddrs;
+}
+
+static char *check_node_announcement_content(struct client *c, const u8 *ann)
+{
+	size_t offset = 66;
```

Comment:
where does this constant come from?

Reply:
Thanks for your review :)
 I noticed [here](https://github.com/ElementsProject/lightning/blob/ba41238df926e2a0e45f503c13674987e914faa2/hsmd/hsmd.c#L1544): 

> 	/* 2 bytes msg type + 64 bytes signature */
	size_t offset = 66;